src/SOS/Strike/strike.cpp

```diff
@@ -1570,6 +1570,27 @@ HRESULT PrintObj(TADDR taObj, BOOL bPrintFields = TRUE)
         }
     }
 
+    // Check for Tracked Type and tagged memory
+    ReleaseHolder<ISOSDacInterface11> sos11;
+    if (SUCCEEDED(g_sos->QueryInterface(__uuidof(ISOSDacInterface11), &sos11)))
+    {
+        CLRDATA_ADDRESS objAddr = TO_CDADDR(taObj);
+        BOOL isTrackedType;
+        BOOL hasTaggedMemory;
+        if (SUCCEEDED(sos11->IsTrackedType(objAddr, &isTrackedType, &hasTaggedMemory)))
```

Comment:
I'm fine with leaving this as-is, but if you want to I'd also be fine collapsing it to a single API:

```
BOOL isTrackedType;
CLRDATA_ADDRESS taggedMemory = NULL;
size_t taggedMemorySizeInBytes = 0;
if (SUCCEEDED(sos11->GetTaggedMemory(objAddr, &isTrackedType, &taggedMemory, &taggedMemorySizeInBytes)))
{
    ExtOut("Tracked Type: %s\n", isTrackedType ? "true" : "false");
    if (taggedMemory != nullptr)
    {
        DMLOut("Tagged Memory: %s (%" POINTERSIZE_TYPE "d(0x%" POINTERSIZE_TYPE "x) bytes)\n",
                    DMLTaggedMemory(taggedMemory), taggedMemorySizeInBytes, taggedMemorySizeInBytes);
    }
}

It appeared the DAC implementation of GetTaggedMemory() is almost identical to the impl for IsTrackedType()

Reply:
I would prefer to have them separate for the following reason. The `IsTrackedType` interrogates the state of the `MethodTable` and checks an `object`'s `SyncBlock`. The `GetTaggedMemory` simply focuses on the `SyncBlock` and doesn't care about the actual state of the `object`. I've observed cases in GC hole investigations where I have an `object` with no `MethodTable` but the `SyncBlock` has data that could be helpful from an interop perspective. Given the relative cost here I would prefer to have these be separate APIs if we want create a new SOS query without requiring a DAC update.